internal/webhooks/clusterclass.go

```diff
@@ -538,33 +539,45 @@ func validateMachineHealthCheckClasses(clusterClass *clusterv1.ClusterClass) fie
 
 	// Validate ControlPlane MachineHealthCheck if defined.
 	if clusterClass.Spec.ControlPlane.MachineHealthCheck != nil {
+		controlPlaneMHCPath := field.NewPath("spec", "controlPlane", "machineHealthCheck")
```

Comment:
Maybe also just fldPath? This variable is scoped to the if anyway (same below for machineDeploymentMHCPath)

But if you prefer to keep it, just keep it :)